### PR TITLE
Roi branch

### PR DESCRIPTION
**Pull request by roiaharon-pepperi**
Added audit_logs endpoint functions.
note that because there are different types of objects, the AuditInfo property is of type 'any'

### CODE REVIEW

File: endpoints/audit-logs.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    constructor(private service: PapiClient, private uuid: string, private linesPart: string = '') {}
+
+    async get() {
+        return await this.service.get(`/audit_logs/${this.uuid}${this.linesPart}`);
+    }
+    async post(body: any = undefined) {
+        return await this.service.post(`/audit_logs/${this.uuid}${this.linesPart}`, body);
+    }
+    lines() {
+        this.linesPart = '/lines';
```
Comment (Chasky-H) [inline endpoints/audit-logs.ts:15]: return this?

File: endpoints/audit-logs.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import Endpoint from '../endpoint';
+import { PapiClient } from '../papi-client';
+
+class AuditLogEndpoint {
+    constructor(private service: PapiClient, private uuid: string, private linesPart: string = '') {}
+
+    async get() {
+        return await this.service.get(`/audit_logs/${this.uuid}${this.linesPart}`);
+    }
+    async post(body: any = undefined) {
```
Comment (Chasky-H) [inline endpoints/audit-logs.ts:11]: Can the body be empty?
Are there any required fields?

File: endpoints/audit-logs.ts (hunk 1 above)
Comment (roiaharon-pepperi) [reply endpoints/audit-logs.ts:15]: yes, I forgot

File: endpoints/audit-logs.ts (hunk 2 above)
Comment (roiaharon-pepperi) [reply endpoints/audit-logs.ts:11]: sorry, copy & paste  mistake

File: endpoints/audit-logs.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import Endpoint from '../endpoint';
+import { PapiClient } from '../papi-client';
+
+class AuditLogEndpoint {
+    constructor(private service: PapiClient, private uuid: string, private linesPart: string = '') {}
+
+    async get() {
+        return await this.service.get(`/audit_logs/${this.uuid}${this.linesPart}`);
+    }
+    async post(body: any) {
```
Comment (Chasky-H) [inline endpoints/audit-logs.ts:11]: Shouldn't this be of type: AuditLog?

Comment (roiaharon-pepperi) [reply endpoints/audit-logs.ts:11]: yes it should.. 👎

File: entities/audit-logs.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+type AuditTypes = 'data' | 'action';
+type AuditStatus = 'Success' | 'Failure' | 'InProgress' | 'InRetry';
+
+export interface AuditLog {
```
Comment (Chasky-H) [inline entities/audit-logs.ts:4]: some of these are optional when posting?

Comment (roiaharon-pepperi) [reply entities/audit-logs.ts:4]: yes, all of them acording to Hadar. made a fix

**Comment by roiaharon-pepperi** [comment]
> What is the difference between posting /audit_logs and posting /audit_logs/uuid/lines or /audit_logs/uuid?

I had a mixup in the enpoint. there is no post for /audit_log/uuid
posting /audit_logs creates a new audit log object 
posting /audit_logs/uuid/lines create a log entry connects to a specific audit object.

**Pull request merged**